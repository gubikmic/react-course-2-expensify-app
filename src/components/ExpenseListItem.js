import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <h3><Link to={"/edit/" + id}>{description}</Link></h3>
     {numeral(amount / 100).format('$0,0.00')} - date: {moment(createdAt).format('MMMM Do, YYYY')}
  </div>
)

export default (ExpenseListItem)