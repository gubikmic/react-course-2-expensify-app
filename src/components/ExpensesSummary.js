import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'

export class ExpensesSummary extends React.Component {
  render() {
    const expenseWord = this.props.expenseCount === 1 ? 'expense' : 'expenses'
    const expenseCount = this.props.expenseCount
    const formattedExpensesTotal = numeral(this.props.expensesTotal / 100).format('$0,0.00')
    return (
      <div>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</div>
    )
  }
}

const mapStateToProps = (state, props) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: visibleExpenses.map( (expense) => expense.amount ).reduce((a, b) => a + b, 0)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)