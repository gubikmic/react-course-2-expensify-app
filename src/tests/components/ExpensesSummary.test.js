import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'

test('should render ExpensesSummary correctly when no expenses are present', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={0} expensesTotal={0} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary correctly with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={9900} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={9999} />)
  expect(wrapper).toMatchSnapshot()
})
