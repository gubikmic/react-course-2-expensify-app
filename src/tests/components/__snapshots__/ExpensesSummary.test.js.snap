// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly when no expenses are present 1`] = `
<div>
  Viewing 
  0
   
  expenses
   totalling 
  $0.00
</div>
`;

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  Viewing 
  2
   
  expenses
   totalling 
  $99.99
</div>
`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling 
  $99.00
</div>
`;
